src/js/base/secure-loader.js

```diff
@@ -1,10 +1,8 @@
-define("pyret-base/js/secure-loader", ["require", "q", "pyret-base/js/runtime-util"], function(rjs, Q, util) {
+define("pyret-base/js/secure-loader", ["q", "pyret-base/js/runtime-util"], function(Q, util) {
   function compileInNewScriptContext(src) {
     var promise = Q.defer();
-    var loader_callback_count = 0;
     if(util.isBrowser()) {
-      loader_callback_count += 1;
-      var callback = "loader_callback_" + loader_callback_count;
+      var callback = "loader_callback_" + Math.floor((Math.random() * 1000));
```

Comment:
This should be the counter version – there's no reason to do this randomly when they are being appended/removed, and sequential numbers are better gensyms.